docker/start-puma.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# ensure we stop on error (-e) and log cmds (-x)
+set -ex
+
+mkdir -p tmp/pids/
+rm -f tmp/pids/*.pid
+
+if [ "$RAILS_ENV" != "development" ]; then
+  USER_DATA=$(curl --fail http://169.254.169.254/latest/user-data || echo "")
+
+  if [ "$USER_DATA" == "EMERGENCY_MODE" ]
+  then
+    git pull
+  fi
+
+  bin/rails db:migrate
```

Comment:
how will this impact on the deployment process? 

Especially considering caesar app goes offline during deploy due to this migrate task on boot and the k8 config.

Related issue in caesar, https://github.com/zooniverse/caesar/pull/921

Reply:
Do you have a suggestion? That PR hasn't been merged and doesn't seem to affect when migrations are run, though I may not be seeing the cascading effects. 

All of our non-Panoptes Rails apps put the migration in some script file or another. FWIW, I'm not overly concerned about a few seconds of downtime during a deploy, but I am of course happy to hear alternatives.

Reply:
On staging it's fine to migrate on boot, on production less so. 

Current plan was to run the migration as a kubernetes job that doesn't retry but does report on failure. this would require some chat ops changes to create the job and apply it but its doable. 
https://cloud.google.com/kubernetes-engine/docs/how-to/jobs
https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/#handling-pod-and-container-failures

short term i'd say keep this in mind and when you are deploying production you're front end may experience service drop outs until we fix this (or you modify the k8 deployment configs to handle pod rotation better)

Reply:
Will keep it in mind